llmfoundry/data/finetuning/tasks.py

```diff
@@ -246,7 +246,15 @@ def _tokenize_prompt_response_formatted_example(
             f'Unable to tokenize example because {response_key} was not a string. {example=}'
         )
 
-    return tokenizer(text=prompt, text_target=response)
+    tokenized_sample = tokenizer(text=prompt, text_target=response)
+
+    # Remove the BOS token from the start of the labels if it was automatically added
+    if hasattr(tokenizer, 'add_bos_token') and tokenizer.add_bos_token:
```

Comment:
we will also need to do this above on L211. pull out into a shared helper fn?

Reply:
ah good catch, thanks. will fix